packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -132,95 +163,133 @@ - (void)initialize:(FlutterError *__autoreleasing *)error {
-    } @catch (NSException *exception) {
-      *error = [FlutterError errorWithCode:@"video_player" message:exception.reason details:nil];
-      return nil;
-    }
+    NSString *assetPath = [self prepareAssetPathFromCreationOptions:options];
+    return [[FVPTextureBasedVideoPlayer alloc] initWithAsset:assetPath
+                                                frameUpdater:frameUpdater
+                                                 displayLink:displayLink
+                                                   avFactory:_avFactory
+                                                   registrar:self.registrar
+                                                  onDisposed:onDisposed];
   } else if (options.uri) {
-    player = [[FVPVideoPlayer alloc] initWithURL:[NSURL URLWithString:options.uri]
-                                    frameUpdater:frameUpdater
-                                     displayLink:displayLink
-                                     httpHeaders:options.httpHeaders
+    return [[FVPTextureBasedVideoPlayer alloc] initWithURL:[NSURL URLWithString:options.uri]
+                                              frameUpdater:frameUpdater
+                                               displayLink:displayLink
+                                               httpHeaders:options.httpHeaders
+                                                 avFactory:_avFactory
+                                                 registrar:self.registrar
+                                                onDisposed:onDisposed];
+  }
+
+  return nil;
+}
+
+- (nullable FVPVideoPlayer *)createPlatformViewPlayerWithOptions:
+    (nonnull FVPCreationOptions *)options {
+  // FVPVideoPlayer contains all required logic for platform views.
+  if (options.asset) {
+    NSString *assetPath = [self prepareAssetPathFromCreationOptions:options];
+    return [[FVPVideoPlayer alloc] initWithAsset:assetPath
                                        avFactory:_avFactory
                                        registrar:self.registrar];
-    return @([self onPlayerSetup:player frameUpdater:frameUpdater]);
-  } else {
-    *error = [FlutterError errorWithCode:@"video_player" message:@"not implemented" details:nil];
-    return nil;
+  } else if (options.uri) {
+    return [[FVPVideoPlayer alloc] initWithURL:[NSURL URLWithString:options.uri]
+                                   httpHeaders:options.httpHeaders
+                                     avFactory:_avFactory
+                                     registrar:self.registrar];
   }
+
+  return nil;
 }
 
-- (void)disposePlayer:(NSInteger)textureId error:(FlutterError **)error {
-  NSNumber *playerKey = @(textureId);
-  FVPVideoPlayer *player = self.playersByTextureId[playerKey];
-  [self.registry unregisterTexture:textureId];
-  [self.playersByTextureId removeObjectForKey:playerKey];
-  if (!player.disposed) {
-    [player dispose];
+- (NSString *)prepareAssetPathFromCreationOptions:(nonnull FVPCreationOptions *)options {
```

Comment:
Nit: `assetPathFromCreationOptions:`

(Similar reason here; since the method is intended to create and return something, the name should generally start by naming that thing, without a verb.)